include/Sema/TypeMapper.hpp

```diff
@@ -0,0 +1,33 @@
+#include "ASTWalker.hpp"
+#include "Types.hpp"
+
+namespace glu::sema {
+
+template <typename Impl>
+class TypeMapper : public glu::ast::ASTWalker<TypeMapper<Impl>, void> {
+    Impl *asImpl() { return static_cast<Impl &>(*this); }
```

Comment:
asImpl is already defined in ASTWalker, but you need to give it the same Impl
```suggestion
class TypeMapper : public glu::ast::ASTWalker<Impl, void> {
```